### DISCUSSION TITLE
Expected response for PropertyDataFetcher(value) of a string

### DISCUSSION THREAD

**Discussion by Simar-Pannu** [graphql-java/graphql-java#3226] [Q&A] [answered]
Is  the expected response from the PropertyDataFetcher when returning property name `value` of an empty string is a empty byte array.

For example this PropertyDataFetchingTest returns true:
```
    def "empty string source property of value is a empty byte array"() {
        def environment = env("")
        def fetcher = new PropertyDataFetcher("value")
        def value = fetcher.get(environment)

        expect:
        Arrays.equals(value, new byte[0])
    }
```
The reason I ask is because the execution result of the byte array will return `value:[B@7ac9af2a]`.

For example this ExecutionStrategyTest will fail as it expects a similar value as above.

```
def "value for a empty string"() {
        given:
        def someFieldName = "value"
        def testTypeName = "Object"
        def fieldDefinition = newFieldDefinition()
                .name(someFieldName)
                .type(GraphQLString)
                .build()
        def objectType = newObject()
                .name(testTypeName)
                .field(fieldDefinition)
                .build()

        def resultP = new ResultPath(new ResultPath(), "getObject")

        ExecutionContext executionContext = buildContext()
        def fieldType = objectType

        def executionStepInfo = ExecutionStepInfo.newExecutionStepInfo().type(fieldType).path(resultP).fieldDefinition(fieldDefinition).build()
        NonNullableFieldValidator nullableFieldValidator = new NonNullableFieldValidator(executionContext, executionStepInfo)
        Field getObject = newField("value")
                .selectionSet(newSelectionSet()
                        .selection(Arrays.asList(new Field("value"))).build())
                        .build()


      //  def result = new byte[0]
        def result = ""
        def parameters = newParameters()
                .executionStepInfo(executionStepInfo)
                .source(result)
                .path(resultP)
                .nonNullFieldValidator(nullableFieldValidator)
                .fields(mergedSelectionSet(['getObject': mergedField([getObject])]))
                .field(mergedField(getObject))
                .build()

        when:
        def executionResult = executionStrategy.completeValue(executionContext, parameters).fieldValue.join()

        then:
        executionResult.data == result
    }
```

**Comment by bbakerman** [comment, marked as answer]
> Is the expected response from the PropertyDataFetcher when returning property name value of an empty string is a empty byte array.

`PropertyDataFetcher` is responsible for return an object from a method or field in the source object.

It NEVER interprets the result ever - other code does - namely the Scalar implementations do.

SO in production (as opposed to your hand crafted tests above) there would be a field and be a source object

If we had a property called `name` then it would try `getName()` method on the source object or `name()` if its a  Java record or `name` if there is a public java field on the object.

But WHATEVER that returns is returned from  `PropertyDataFetcher` unchanged.  This would then be passed to to the scalar code t interpret it  (this is called completing the field)

The current String scalar will call `String.valueOf(x)` so if you return a byte array - then it will be the `String.valueOf` it

(Future string scalars will not allow any object - but thats for a future release )

**Comment by bbakerman** [comment]
we do not have any code that turns byte arrays into strings in graphql-java nor in https://github.com/graphql-java/graphql-java-extended-scalars/

So I think using byte arrays wont work
